### ISSUE TITLE
Tests fail with Conda env Installation

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#450]
Following this procedure results in an installation of ray that won't pass the given tests: 

Creating a new EC2 Environment (Ubuntu Server 14.04, m4.10xlarge)
1. Curl miniconda script from online
2. Install miniconda
3. Create a new conda environment with python=3.5 and activate
4. Install ray prereqs
5. Install ray
6. Run tests

The following tests fail: 
```
======================================================================
FAIL: testLoadBalancing (__main__.SchedulingAlgorithm)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/runtest.py", line 1341, in testLoadBalancing
    self.attempt_to_load_balance(f, [], 100, num_local_schedulers, 25)
  File "test/runtest.py", line 1326, in attempt_to_load_balance
    self.assertLess(attempts, num_attempts)
AssertionError: 20 not less than 20
======================================================================
FAIL: testLoadBalancingWithDependencies (__main__.SchedulingAlgorithm)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/runtest.py", line 1352, in testLoadBalancingWithDependencies
    num_local_schedulers=num_local_schedulers)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1017, in _init
    worker=global_worker, actor_id=NIL_ACTOR_ID)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1345, in connect
    assert not worker.connected, error_message
AssertionError: Perhaps you called ray.init twice by accident?
======================================================================
FAIL: testPassingArgumentsByValue (__main__.SerializationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/runtest.py", line 191, in testPassingArgumentsByValue
    ray.init(num_workers=1)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1063, in init
    num_gpus=num_gpus)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1017, in _init
    worker=global_worker, actor_id=NIL_ACTOR_ID)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1345, in connect
    assert not worker.connected, error_message
AssertionError: Perhaps you called ray.init twice by accident?
======================================================================
FAIL: testRecursiveObjects (__main__.SerializationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/runtest.py", line 160, in testRecursiveObjects
    ray.init(num_workers=0)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1063, in init
    num_gpus=num_gpus)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1017, in _init
    worker=global_worker, actor_id=NIL_ACTOR_ID)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1345, in connect
    assert not worker.connected, error_message
AssertionError: Perhaps you called ray.init twice by accident?
======================================================================
FAIL: testCopyingDirectory (__main__.UtilsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/runtest.py", line 974, in testCopyingDirectory
    ray.init(num_workers=1)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1063, in init
    num_gpus=num_gpus)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1017, in _init
    worker=global_worker, actor_id=NIL_ACTOR_ID)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1345, in connect
    assert not worker.connected, error_message
AssertionError: Perhaps you called ray.init twice by accident?
======================================================================
FAIL: testBlockingTasks (__main__.WorkerPoolTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/runtest.py", line 1279, in testBlockingTasks
    ray.init(num_workers=1)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1063, in init
    num_gpus=num_gpus)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1017, in _init
    worker=global_worker, actor_id=NIL_ACTOR_ID)
  File "/home/ubuntu/.local/lib/python3.5/site-packages/ray-0.0.1-py3.5.egg/ray/worker.py", line 1345, in connect
    assert not worker.connected, error_message
AssertionError: Perhaps you called ray.init twice by accident?
----------------------------------------------------------------------
Ran 32 tests in 63.763s
FAILED (failures=6)
```

**Comment by pcmoritz** [comment]
Hm, I couldn't reproduce it, here are the steps I tried:

```
wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
bash Miniconda3-latest-Linux-x86_64.sh
conda create python=3.5 --name rayenv
source activate rayenv
sudo apt-get update
sudo apt-get install -y git cmake build-essential autoconf curl libtool libboost-all-dev unzip
pip install numpy cloudpickle funcsigs colorama psutil redis flatbuffers
git clone https://github.com/ray-project/ray
cd ray/python
python setup.py install
cd ..
python test/runtest.py
```

**Comment by richardliaw** [comment]
After `pip install ...` I run `conda install libgcc` also

**Comment by robertnishihara** [comment]
One possibility is that the problem was introduced in #430. We used to start 10 workers by default, now we're starting `psutil.num_cpus()` workers, which in your case is something like 40, so that's probably throwing off the load balancing test since more things can get scheduled locally before getting passed on to the global scheduler.

How about we set `num_cpus=10` explicitly in this test?

**Comment by robertnishihara** [comment]
Does #452 address the problem you were seeing?

**Comment by richardliaw** [comment]
Yep it does.

**Comment by robertnishihara** [comment]
Addressed by #452.
